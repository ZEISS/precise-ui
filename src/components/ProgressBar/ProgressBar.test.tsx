import * as React from 'react';
import * as enzyme from 'enzyme';
import { ProgressBar } from './';

describe('<ProgressBar />', () => {
  it('should render empty default <ProgressBar /> element', () => {
    const wrapper = enzyme.shallow(<ProgressBar />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the <ProgressBar /> with 50%', () => {
    const wrapper = enzyme.shallow(<ProgressBar value={50} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the <ProgressBar /> with 50% with custom boundaries', () => {
    const wrapper = enzyme.shallow(<ProgressBar value={15} maximum={20} minimum={10} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the <ProgressBar /> with 0% with custom boundaries', () => {
    const wrapper = enzyme.shallow(<ProgressBar value={15} maximum={20} minimum={15} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the <ProgressBar /> with 100% with custom boundaries', () => {
    const wrapper = enzyme.shallow(<ProgressBar value={20} maximum={20} minimum={15} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the <ProgressBar /> with a title', () => {
    const wrapper = enzyme.shallow(<ProgressBar title="My Progress" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the <ProgressBar /> with a description', () => {
    const wrapper = enzyme.shallow(<ProgressBar description="Please wait until we finish the work" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the <ProgressBar /> with a title, description, at 75%', () => {
    const wrapper = enzyme.shallow(<ProgressBar value={75} title="My Progress" description="My Description" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the <ProgressBar /> animated at 75%', () => {
    const wrapper = enzyme.shallow(<ProgressBar value={75} animate />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the <ProgressBar /> with maximum and minimum', () => {
    const wrapper = enzyme.shallow(<ProgressBar maximum={75} minimum={25} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should cut the maximum at 100 percent', () => {
    const wrapper = enzyme.shallow(<ProgressBar maximum={10} value={25} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the <ProgressBar /> with maximum and minimum and some stops', () => {
    const wrapper = enzyme.shallow(<ProgressBar maximum={75} minimum={25} stops={[30, 50, 70]} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the <ProgressBar /> with cut stops', () => {
    const wrapper = enzyme.shallow(<ProgressBar stops={[-10, 0, 50, 100, 110]} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the <ProgressBar /> with a single stop', () => {
    const wrapper = enzyme.shallow(<ProgressBar stops={[0.5]} maximum={1} />);
    expect(wrapper).toMatchSnapshot();
  });
});
