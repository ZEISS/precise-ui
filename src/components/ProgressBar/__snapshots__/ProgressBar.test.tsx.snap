// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProgressBar /> should cut the maximum at 100 percent 1`] = `
<styled.div>
  <styled.div
    type="primary"
  >
    <styled.div
      style={
        Object {
          "transition": "none",
          "width": "100%",
        }
      }
    />
  </styled.div>
</styled.div>
`;

exports[`<ProgressBar /> should render empty default <ProgressBar /> element 1`] = `
<styled.div>
  <styled.div
    type="primary"
  >
    <styled.div
      style={
        Object {
          "transition": "none",
          "width": "0%",
        }
      }
    />
  </styled.div>
</styled.div>
`;

exports[`<ProgressBar /> should render the <ProgressBar /> animated at 75% 1`] = `
<styled.div>
  <styled.div
    type="primary"
  >
    <styled.div
      style={
        Object {
          "transition": "width 200ms",
          "width": "75%",
        }
      }
    />
  </styled.div>
</styled.div>
`;

exports[`<ProgressBar /> should render the <ProgressBar /> with 0% with custom boundaries 1`] = `
<styled.div>
  <styled.div
    type="primary"
  >
    <styled.div
      style={
        Object {
          "transition": "none",
          "width": "0%",
        }
      }
    />
  </styled.div>
</styled.div>
`;

exports[`<ProgressBar /> should render the <ProgressBar /> with 50% 1`] = `
<styled.div>
  <styled.div
    type="primary"
  >
    <styled.div
      style={
        Object {
          "transition": "none",
          "width": "50%",
        }
      }
    />
  </styled.div>
</styled.div>
`;

exports[`<ProgressBar /> should render the <ProgressBar /> with 50% with custom boundaries 1`] = `
<styled.div>
  <styled.div
    type="primary"
  >
    <styled.div
      style={
        Object {
          "transition": "none",
          "width": "50%",
        }
      }
    />
  </styled.div>
</styled.div>
`;

exports[`<ProgressBar /> should render the <ProgressBar /> with 100% with custom boundaries 1`] = `
<styled.div>
  <styled.div
    type="primary"
  >
    <styled.div
      style={
        Object {
          "transition": "none",
          "width": "100%",
        }
      }
    />
  </styled.div>
</styled.div>
`;

exports[`<ProgressBar /> should render the <ProgressBar /> with a description 1`] = `
<styled.div>
  <styled.div
    type="primary"
  >
    <styled.div
      style={
        Object {
          "transition": "none",
          "width": "0%",
        }
      }
    />
  </styled.div>
  <styled.div>
    Please wait until we finish the work
  </styled.div>
</styled.div>
`;

exports[`<ProgressBar /> should render the <ProgressBar /> with a single stop 1`] = `
<styled.div>
  <styled.div
    type="primary"
  >
    <styled.div
      style={
        Object {
          "transition": "none",
          "width": "0%",
        }
      }
    />
    <styled.div
      active={false}
      key="0-at-0.5"
      style={
        Object {
          "left": "50%",
        }
      }
    />
  </styled.div>
</styled.div>
`;

exports[`<ProgressBar /> should render the <ProgressBar /> with a title 1`] = `
<styled.div>
  <styled.div>
    My Progress
  </styled.div>
  <styled.div
    type="primary"
  >
    <styled.div
      style={
        Object {
          "transition": "none",
          "width": "0%",
        }
      }
    />
  </styled.div>
</styled.div>
`;

exports[`<ProgressBar /> should render the <ProgressBar /> with a title, description, at 75% 1`] = `
<styled.div>
  <styled.div>
    My Progress
  </styled.div>
  <styled.div
    type="primary"
  >
    <styled.div
      style={
        Object {
          "transition": "none",
          "width": "75%",
        }
      }
    />
  </styled.div>
  <styled.div>
    My Description
  </styled.div>
</styled.div>
`;

exports[`<ProgressBar /> should render the <ProgressBar /> with cut stops 1`] = `
<styled.div>
  <styled.div
    type="primary"
  >
    <styled.div
      style={
        Object {
          "transition": "none",
          "width": "0%",
        }
      }
    />
    <styled.div
      active={true}
      key="0-at--10"
      style={
        Object {
          "left": "0%",
        }
      }
    />
    <styled.div
      active={true}
      key="1-at-0"
      style={
        Object {
          "left": "0%",
        }
      }
    />
    <styled.div
      active={false}
      key="2-at-50"
      style={
        Object {
          "left": "50%",
        }
      }
    />
    <styled.div
      active={false}
      key="3-at-100"
      style={
        Object {
          "left": "100%",
        }
      }
    />
    <styled.div
      active={false}
      key="4-at-110"
      style={
        Object {
          "left": "100%",
        }
      }
    />
  </styled.div>
</styled.div>
`;

exports[`<ProgressBar /> should render the <ProgressBar /> with maximum and minimum 1`] = `
<styled.div>
  <styled.div
    type="primary"
  >
    <styled.div
      style={
        Object {
          "transition": "none",
          "width": "0%",
        }
      }
    />
  </styled.div>
</styled.div>
`;

exports[`<ProgressBar /> should render the <ProgressBar /> with maximum and minimum and some stops 1`] = `
<styled.div>
  <styled.div
    type="primary"
  >
    <styled.div
      style={
        Object {
          "transition": "none",
          "width": "0%",
        }
      }
    />
    <styled.div
      active={false}
      key="0-at-30"
      style={
        Object {
          "left": "10%",
        }
      }
    />
    <styled.div
      active={false}
      key="1-at-50"
      style={
        Object {
          "left": "50%",
        }
      }
    />
    <styled.div
      active={false}
      key="2-at-70"
      style={
        Object {
          "left": "90%",
        }
      }
    />
  </styled.div>
</styled.div>
`;
